api/src/main/java/jakarta/config/Config.java

```diff
@@ -30,27 +28,73 @@
  */
 public interface Config {
     /**
-     * Return all the currently registered {@link jakarta.config.spi.ConfigSource sources} for this configuration.
+     * Context related to the root configuration instance.
+     *
+     * @return configuration context
+     */
+    ConfigContext context();
+
+    /**
+     * Fully qualified key of this config node (such as {@code server.port}).
+     * Returns an empty String for root config.
+     *
+     * @return key of this config
+     */
+    String getKey();
+
+    /**
+     * Name of this node - the last element of a fully qualified key.
      * <p>
-     * The returned sources will be sorted by priority and name, which can be iterated in a thread-safe
-     * manner. The {@link java.lang.Iterable Iterable} contains a fixed number of {@link jakarta.config.spi.ConfigSource
-     * configuration
-     * sources}, determined at application start time, and the config sources themselves may be static or dynamic.
+     * For example for key {@code server.port} this method would return {@code port}.
+     *
+     * @return name of this node
+     */
+    String getName();
+
+    /**
+     * Single sub-node for the specified sub-key.
+     * For example if requested for key {@code server}, this method would return a config
+     * representing the {@code server} node, which would have for example a child {@code port}.
+     * The sub-key can return more than one level of nesting (e.g. using {@code server.tls} would
+     * return a node that contains the TLS configuration under {@code server} node).
+     *
+     * @param key sub-key to retrieve nested node.
+     * @return sub node, never null
+     */
+    Config get(String key);
+
+    /**
+     * Typed value created using a converter function.
+     * The converter is called only if this config node exists.
+     *
+     * @param converter to create an instance from config node
+     * @param <T> type of the object
+     * @return converted value of this node, or an empty optional if this node does not exist
+     * @throws java.lang.IllegalArgumentException if this config node cannot be converted to the desired type
+     */
+    <T> Optional<T> as(Function<Config, T> converter);
+
+    /**
+     * Typed value created using a discovered/built-in converter.
      *
-     * @return the configuration sources
+     * @param type class to convert to
+     * @param <T> type of the object
+     * @return converted value of this node, or an empty optional if this node does not exist
+     * @throws java.lang.IllegalArgumentException if this config node cannot be converted to the desired type
      */
-    Iterable<ConfigSource> getConfigSources();
+    <T> Optional<T> as(Class<T> type);
```

Comment:
I think it is much better to have getValue(String key, Class<T> type) instead of doing get(key).as(type).

Reply:
Convenience method added.